app/src/main/java/org/mozilla/fenix/home/intent/FennecPinnedShortcutsIntentProcessor.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Intent
+import android.content.Intent.ACTION_VIEW
+import androidx.annotation.VisibleForTesting
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.feature.intent.ext.putSessionId
+import mozilla.components.feature.intent.processing.IntentProcessor
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.utils.toSafeIntent
+
+/**
+ * Legacy processor for pinned shortcuts created by Fennec.
+ * https://developer.android.com/guide/topics/ui/shortcuts/creating-shortcuts#pinned
+ */
+class FennecPinnedShortcutsIntentProcessor(
```

Comment:
Since theses shortcuts are distinct from PWA shortcuts (which are also pinned), maybe the class could be called `FennecBookmarkShortcutsIntentProcessor`.

Reply:
Sure. Thanks!